src/NHSD.BuyingCatalogue.Identity.Api/ViewModels/Account/ResetPasswordViewModel.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.ViewModels.Account
+{
+    public sealed class ResetPasswordViewModel
+    {
+        [DataType(DataType.Password)]
+        [PasswordValidation(ErrorMessage = ErrorMessages.PasswordConditionsNotMet)]
+        [DisplayName("Enter a password")]
+        public string Password { get; set; }
+
+        [DataType(DataType.Password)]
+        [Compare("Password", ErrorMessage = ErrorMessages.PasswordMismatch)]
+        [DisplayName("Confirm password")]
+        public string ConfirmPassword { get; set; }
+
+        internal static class ErrorMessages
+        {
+            internal const string PasswordConditionsNotMet = "The password you’ve entered does not meet the criteria";
+            internal const string PasswordMismatch = "Passwords do not match";
+        }
+
+        [AttributeUsage(AttributeTargets.Property, AllowMultiple = true, Inherited = false)]
+        internal class PasswordValidation : ValidationAttribute
+        {
+            public override bool IsValid(object value)
+            {
+                if (!(value is string))
+                {
+                    return false;
+                }
+
+                var specialCharacters = "!@#$%^&*";
+
+                var password = value as string;
```

Comment:
This typecast will always succeed due to the earlier check so we don't need to use `as` here, `is` will be fine.

Another option would be replace the initial check with the `as` form and `return false` if `null`.